### PR TITLE
Sort User By Name

### PR DESCRIPTION
**Pull request by at-nale**
## Description
Sort User By Name
## Reference Requirements
- Task ID: #69

## Reference Isuees
#{issue_id}

## Migrations
 NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
```
 explain select * from `users` where `users`.`deleted_at` is null order by `users`.`name` desc, `name` desc limit 15 offset 0\G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: users
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: 10
     filtered: 10.00
        Extra: Using where; Using filesort
1 row in set, 1 warning (0.01 sec)
```
- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Http/Controllers/Admin/UserController.php [hunk 1]
```diff
@@ -35,9 +35,9 @@ public function index(Request $request)
             $users = User::where("name", "LIKE", "%$keyword%")
                         ->orWhere("email", "LIKE", "%$keyword%");
             Session::flash('message_search', trans('search.message', ['number' => $users->count()]));
-            $users = $users->paginate()->appends(['search' => $keyword]);
+            $users = $users->sortable()->orderby('name', 'desc')->paginate()->appends(['search' => $keyword]);
```
Comment (at-haphan) [inline app/Http/Controllers/Admin/UserController.php]: Sửa lại orderBy

File: app/Http/Controllers/Admin/UserController.php [hunk 2]
```diff
@@ -35,9 +35,9 @@ public function index(Request $request)
...
         } else {
-            $users = User::paginate();
+            $users = User::sortable()->orderby('name', 'desc')->paginate();
```
Comment (at-haphan) [inline app/Http/Controllers/Admin/UserController.php]: sửa lại

File: app/Models/User.php [hunk 3]
```diff
@@ -6,10 +6,11 @@
 use Illuminate\Database\Eloquent\SoftDeletes;
 use Illuminate\Foundation\Auth\User as Authenticatable;
 use Illuminate\Support\Facades\Hash;
+use Kyslik\ColumnSortable\Sortable;
 
 class User extends Authenticatable
 {
-    use SoftDeletes, Notifiable;
+    use SoftDeletes, Notifiable,sortable;
```
Comment (at-haphan) [inline app/Models/User.php]: là Sortable

**Comment by at-haphan** [comment]
LGTM

**Comment by at-vietpham** [comment]
LGTM 🐙

**Pull request merged**